src/DotVVM.Framework/Compilation/Javascript/ParametrizedCode.cs

```diff
@@ -58,18 +75,38 @@ public string ToString(Func<object, CodeParameterAssignment> parameterAssignment
                     sb.Append(stringParts[++i]);
                 }
             }
-            return sb.ToString();
+            var result = sb.ToString();
+            if (allIsDefault)
+                this.evaluatedDefault = result;
+            return result;
+        }
+
+        [Obsolete("ParametrizedCode.ToString use is discouraged, this overload does not return the code, please use the ToString(Func<...> parameterAssigner) overload or ToDefaultString method. Note that these may return an exception.", true)]
+        public override string ToString()
+        {
+            // leave for debug purposes.
+            return base.ToString();
+        }
+
+        public string ToDefaultString()
+        {
+            if (this.evaluatedDefault != null)
+                return this.evaluatedDefault;
+            return ToString(_ => default);
         }
 
         /// <summary>
         /// Assigns parameters and return new ParametrizedCode. If parameter is not assigned, it is copied to the resulting parameter. Assigner can also replace parameter by script that contains another parameters.
         /// </summary>
         public ParametrizedCode AssignParameters(Func<object, CodeParameterAssignment> parameterAssignement)
```

Comment:
I think you should change argument type to `CodeSymbolicParamerer` (btw: typo in name).

Reply:
BTW: Is there a nice way to find out what parameters were actually assigned? I can do it in `parameterAssignement` (typo in name, should be `parameterAssignment` without `e`) but it isn't really nice.

Reply:
No, there is no way to get assigned parameters, they are wiped away when assigned (only the resulting string is left). If you need more powerful API you will need the Javascript AST

Reply:
Ok. I've realized I can look what params are defined in `ParametrizedCode` before assigning them.